primo/demo/PRIMO - Example_1.ipynb

```diff
@@ -5242,86 +5667,748 @@
+       "      <td>55</td>\n",
+       "      <td>5</td>\n",
+       "      <td>62.20</td>\n",
+       "      <td>40.89</td>\n",
+       "    </tr>\n",
+       "  </tbody>\n",
+       "</table>\n",
+       "</div>"
+      ],
+      "text/plain": [
+       " Project  Number of Wells  Impact Score [0-100]  Efficiency Score [0-100]\n",
+       "      75                5                 63.23                     57.24\n",
+       "      62               11                 65.45                     51.32\n",
+       "      87                5                 64.67                     50.06\n",
+       "     100                5                 69.35                     49.27\n",
+       "      72                7                 64.83                     48.62\n",
+       "      58                8                 65.85                     44.21\n",
+       "      52                5                 64.17                     43.99\n",
+       "      55                5                 62.20                     40.89"
+      ]
+     },
+     "metadata": {},
+     "output_type": "display_data"
+    },
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Total budget recommended for plugging oil wells is: 4,284,000\n"
+     ]
+    }
+   ],
+   "source": [
+    "violation_oil = Recalculate(selected_oil, well_add_list_oil, well_remove_list_oil,oil_gdf,opt_inputs_oil)\n",
+    "\n",
+    "violate_cost_oil=violation_oil.assess_budget()\n",
+    "violate_operator_oil=violation_oil.assess_owner_well_count()\n",
+    "violate_distance_oil=violation_oil.assess_distances()\n",
+    "\n",
+    "if violate_cost_oil > 0:\n",
+    "    display(Markdown(f\"\"\" After the modification, the total budget is over the limit by ${int(violate_cost_oil)}.\n",
+    "                     Please consider to modify wells you have selected use the widget above or re-run the optimization problem.\"\"\"))\n",
+    "\n",
+    "if bool(violate_operator_oil) is not False:\n",
+    "    display(Markdown(f\"\"\" After the modification, the following owners have more than {max_wells_per_owner} well(s) \n",
+    "                     being selected. Please consider to modify wells you have selected use the widget above\n",
+    "                     or re-run the optimization problem.\"\"\"))\n",
+    "    violate_operator_oil_df = pd.DataFrame.from_dict(violate_operator_oil,orient='index')\n",
+    "    violate_operator_oil_df = violate_operator_oil_df.reset_index()\n",
+    "    violate_operator_oil_df = violate_operator_oil_df.rename(columns={'index': 'Owner', 0: 'Number of wells', 1: 'Wells'})\n",
+    "    display(violate_operator_oil_df)\n",
+    "\n",
+    "\n",
+    "if bool(violate_distance_oil) is not False:\n",
+    "    display(Markdown(f\"\"\" After the modification, the following projects have wells are far away from each others. \n",
+    "                     Please consider to modify wells you have selected use the widget above\n",
```

Comment:
Same correction here: "Please consider modifying the wells you have selected by using either the widget above, or by re-running the optimization problem"

Reply:
Fixed